src/monattack.cpp

```diff
@@ -1418,74 +1418,54 @@ void mattack::fungus(monster *z, int index)
     z->moves -= 200;   // It takes a while
     z->reset_special(index); // Reset timer
     if (g->u.has_trait("THRESH_MYCUS")) {
-        z->friendly = 1;
+        z->friendly = 100;
     }
-    monster spore(GetMType("mon_spore"));
-    int mondex;
     //~ the sound of a fungus releasing spores
     sounds::sound(z->pos(), 10, _("Pouf!"));
     if (g->u.sees( *z )) {
         add_msg(m_warning, _("Spores are released from the %s!"), z->name().c_str());
     }
-    for (int i = -1; i <= 1; i++) {
-        for (int j = -1; j <= 1; j++) {
-            if (i == 0 && j == 0) {
+
+    // Use less laggy methods of reproduction when there is a lot of mons around
+    double spore_chance = 0.25;
+    int range = 1;
+    if( g->num_zombies() > 25 ) {
+        // Number of monsters in the bubble and the resulting average number of spores per "Pouf!":
+        // 0-25: 2
+        // 50  : 0.5
+        // 75  : 0.22
+        // 100 : 0.125
+        // Assuming all monsters in the bubble were fungaloids (unlikely), the average number of spores per generation:
+        // 25  : 50
+        // 50  : 25
+        // 75  : 17
+        // 100 : 13
+        spore_chance *= ( 25.0 / g->num_zombies() ) * ( 25.0 / g->num_zombies() );
+        if( x_in_y( g->num_zombies(), 100 ) ) {
+            range++;
+            // Don't make the increased range spawn more spores
+            const double old_area = ( ( range - 1 ) * ( range - 1 ) ) - 1;
+            const double new_area = ( range * range ) - 1;
+            spore_chance *= old_area / new_area;
```

Comment:
Fixed the math.

I added the default parameters mostly because `create_spores` had one.
Currently they are unused, but could simplify writing some functions later (some sort of spore rain event etc.).